utils.sh

```diff
@@ -99,6 +99,35 @@ function create_cluster() {
 
     trap auth_template_and_removetmp EXIT
     $OPENSHIFT_INSTALLER --dir "${assets_dir}" --log-level=debug create cluster
+
+    # Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1816904
+    wait_for_cluster
+}
+
+wait_for_cluster() {
+    TIMEOUT_MINUTES=30
+    if ! timeout ${TIMEOUT_MINUTES}m bash <<EOF
+while (( $(oc get nodes -l node-role.kubernetes.io/worker='' -o name | wc -l) < $NUM_WORKERS )); do
+  echo "Waiting for $NUM_WORKERS workers"
+  sleep 5
+done
+EOF
+    then
+      echo "Error - timeout waiting for $NUM_WORKERS workers"
+      exit 1
+    fi
+
+    TIMEOUT_MINUTES=15
+    for worker in $(oc get nodes -l node-role.kubernetes.io/worker='' -o name); do
+        echo "$worker registered, waiting $TIMEOUT_MINUTES minutes for Ready condition ..."
+        oc wait $worker --for=condition=Ready --timeout=$[${TIMEOUT_MINUTES} * 60]s
+    done
+
+    if [ "${OPENSHIFT_CI}" == true ]; then
```

Comment:
I said marketplace operator is in degraded state in disconnected but I was wrong. Actually marketplace operator is available state and the pods it created are in erroneous state. We can remove this check from here.